custom_components/shopping_list_with_grocy/__init__.py

```diff
@@ -129,6 +129,31 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
 
     return True
 
+    #
+    # To v4
+    #
+
+    if config_entry.version == 3:
+        v2_options = dict(config_entry.options)
+        v2_data = dict(config_entry.data)
+        if v2_options('adding_images', FALSE) is TRUE:
+            v2_options['image_download_size'] = 100
+            v2_data['image_download_size'] = 100
+        else:
+            v2_options['image_download_size'] = 0
+            v2_data['image_download_size'] = 0
+
+        v2_options.pop('adding_images', NONE)
```

Comment:
if `adding_images` also exists in v2_data, it should also be removed from there (options are when config is changed AFTER plugin init, data if for options selected during plugin init)

Reply:
done